### PR TITLE
upgrade kubeclipper platform or component

### LINKED ISSUE #91: upgrade kubeclipper platform or component
**What's it about?**

**What's the reason why we need it?**

I believe this is an important feature for KubeClipper. There're a few use cases:

* case one
* case two
* ...

Please leave your comments below if there's anyone agrees with me. Or just give me a thumb up.

**Area Suggestion**

/kind feature-request

### PR DESCRIPTION
**Pull request by Xvv-v**
Signed-off-by: Xvv-v <xu.jingwen@99cloud.net>

### What type of PR is this?
/kind feature

### Which issue(s) this PR fixes:

Fixes #91 

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note

```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by x893675** [comment]
/cc @lixd 
help review

File: pkg/cli/upgrade/upgrade.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+		err = o.replaceService(o.component)
+	}
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *UpgradeOptions) sendPackage() error {
+	tar := fmt.Sprintf("rm -rf %s && tar -xvf %s -C %s",
```
Comment (lixd) [inline pkg/cli/upgrade/upgrade.go:168]: maybe need a pkg format check. refer ResourcePush func in resource.go

File: pkg/cli/upgrade/upgrade.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+			continue
+		}
+		if err := o.replaceService(component); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (o *UpgradeOptions) copyCmd(component string) (cp string) {
```
Comment (lixd) [inline pkg/cli/upgrade/upgrade.go:205]: backup old resource first better.

File: pkg/cli/upgrade/upgrade.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+	}
+	if fileutil.IsDir(o.pkg) {
+		return fmt.Errorf("--pkg can't be a directory")
+	}
+	return nil
+}
+
+func (o *UpgradeOptions) Validate(cmd *cobra.Command, args []string) error {
+	if len(args) < 1 {
+		return utils.UsageErrorf(cmd, "You must specify the component of kubeclipper to upgrade, support [ agent | server | etcd | console ] now")
```
[the remark this answers is not in the document]
Comment (Xvv-v) [reply pkg/cli/upgrade/upgrade.go:120]: map的值用来存nodeip，所以没用sets.String

**Review by lixd — APPROVED** [review]
/lgtm

**Review by lixd — APPROVED** [review]
/lgtm

**Pull request merged**